src/corgi-tag-watcher.js

```
@@ -0,0 +1,82 @@
+    const contentRequest = context.repo({ path: 'META-INF/books.xml' })
+    const contentMetadata = await context.octokit.repos.getContent(contentRequest)
+    const content = Buffer.from(contentMetadata.data.content, contentMetadata.data.encoding)
+
+    const repo = context.payload.repository.name
+
+    const books = []
+    const parser = sax.parser()
+    parser.onopentag = (node) => {
+      if (node.name === 'BOOK') {
+        books.push(node.attributes.SLUG)
+      }
+    }
+    parser.write(content).close()
+
+    // send job details to CORGI API
+    // make request per book + job type
+    let jobStatus = 'failed to queue'
+
+    try {
+      for (const slug of books) {
+        for (const jobType of [3, 4]) {
+          // 3: git-pdf
+          // 4: git-distribution-preview
+
+          logger.info(`collection_id: ${repo}/${slug}`)
+          logger.info(`job_type_id: ${jobType}`)
+
+          const payload = {
+            collection_id: `${repo}/${slug}`,
+            job_type_id: jobType,
+            status_id: 1,
+            version: `${context.payload.ref}`,
+            style: 'business-ethics' // TODO: add style to META-INF
+          }
+
+          const response = await fetch(CORGI_URL, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(payload)
+          })
+          logger.info(response.status.toString())
+          if (response.status !== 200) { throw new Error('CORGI is not responding!') }
+        }
+      }
+
+      jobStatus = 'successfully queued'
+    } catch { }
+
+    let bookList = ''
+    books.forEach((book) => {
+      bookList += `\n - ${book}`
+    })
+
+    logger.info('Attempting to send slack message to channel')
+    await fetch(SLACK_URL, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+
+      body: JSON.stringify({
+        text:
+          `CORGI job(s) ${jobStatus} for ${bookList}`
```

If the first few jobs POST okay but then there's a failure, would all jobs end up being reported as failed? (perhaps that's close enough reporting for our purposes since it is somewhat all or nothing, so mostly just asking to confirm)

---

I believe so. It should throw an error if any of them fail.